claat/proto-renderer/repeated_utils.go

```diff
@@ -0,0 +1,54 @@
+package genrenderer
+
+import (
+	"fmt"
+	"text/template"
+
+	"github.com/googlecodelabs/tools/third_party"
+)
+
+// RenderRepeated returns iterator-friend string outputs of the passed
+// repeated proto by recursively rendering their contents.
+func RenderRepeated(elSlice interface{}, t *template.Template) []string {
+	contents := AssertRepeated(elSlice)
+	sz := len(contents)
+	renderedEls := make([]string, sz)
+
+	if sz < 1 {
+		// debug-friendly panic
+		panic(fmt.Sprintf("RenderRepeated empty repeated field %#v", contents))
+	}
+
+	for i := 0; i < sz; i++ {
+		// Recursive rendering happens here
+		renderedEls[i] = ExecuteTemplate(contents[i], t)
+	}
+
+	return renderedEls
+}
+
+// AssertRepeated turns a generic proto slice into typed-slice that can be
+// interated over without reflection. Panics if the passed type is not
+// explicitly defined
+func AssertRepeated(el interface{}) (guaranteedProtoSlice []interface{}) {
```

Comment:
Same comment as other PR - this "assert" method is doing more than just asserting something.

Reply:
Per https://github.com/googlecodelabs/tools/pull/291#discussion_r297828437, suggestions on function names besides `AssertUnderlyingRepeated` welcomed.  This is a type-assertion wrapper

Reply:
and in addition to the above comment, since the use of reflection was dropped, merging these two functions will result in an undable mess. Rather have a small, easy to read logic + a smaller, decoupled mess.

only downside here is that, if wanted, future concurrency will need to implemented in two places

Reply:
I guess what I'm trying to say is conventionally methods called "assert" do one of two things:

a) nothing (e.g. assertTrue(true) will do nothing)
b) raise an exception (e.g. assertTrue(false) will throw an error)

In this case we have an AssertX method that takes in a value and then gives you a different value in return. Conventional assert methods don't do that.

Reply:
Oh I see. Yeah coming from a testing-framework background this makes sense. I originally named these `TypeAssertX` but I thought "type" was redundant.

This method are doing the same functionality as your point #2, `AssertType(notType)` => `panic` (Golang's hard exceptions). I'll do research to see if point #1 _actually_ doesn't return anything.

At any rate, "Type asserts" are very much a thing (link TBA, check comment in a bit).

Reply:
type assert link included